### PR TITLE
Add codeowners file

### PR DESCRIPTION
**Pull request by forrestxia**
##### SUMMARY
Add CODEOWNERS file

Relates OR Closes #0000

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

#### Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

#### Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

#### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Review by MangirishK — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# documentation for this file can be found at:
+# https://help.github.com/en/articles/about-code-owners
+
+# These are the default owners for the code and will
+# be requested for review when someone opens a pull request.
+# Order is alphabetical for easier maintenance.
+#
+# All Terraform Engineers in Dell (dell/TerraformDell)
+
+# for all files:
```
Comment (gokul-srivathsan) [inline .github/CODEOWNERS:12]: Is it the correct DL?

Comment (forrestxia) [reply .github/CODEOWNERS:12]: @gokul-srivathsan It intents to use github team as the code owners. @org/teamname is the syntax of referenecing a team in codeowners file.

**Review by sapana05 — APPROVED** [review]

**Review by baoy1 — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (baoy1) [reply .github/CODEOWNERS:12]: Weird. Github shows
> Unknown owner on line 13: make sure the team @dell/TerraformDell exists, is publicly visible, and has write access to the repository
* @dell/TerraformDell

Comment (forrestxia) [reply .github/CODEOWNERS:12]: Add write access to terraformdell team that removes the warning.

**Pull request merged**